### ISSUE TITLE
Hyperparameter search class

### ISSUE THREAD

**Issue by evanfeinberg** [deepchem/deepchem#89]
Searching for hyperparameters is both very easy/straightforward with the latest build, especially w/ #87 . However, I'm inclined to think there should be a HyperparameterSearch class or equivalent where the user can specify the names and values of the hyperparameters over which to search, and the best model on the validation set is returned. This should eliminate a lot of really repetitive code writing, since it all falls under a very similar pattern.

**Comment by rbharath** [comment]
Added by #99.
